server/auth/registrar.go

```diff
@@ -25,7 +30,344 @@ var (
+	}
+
+	dbBond := &db.Bond{
+		Version:  postBond.BondVer,
+		AssetID:  assetID,
+		CoinID:   bondCoinID,
+		Script:   postBond.BondScript,
+		Amount:   amt,
+		LockTime: lockTime,
+	}
+
+	// Either store the bond or start a block waiter to activate the bond and
+	// respond with a PostBondResult when it is fully-confirmed.
+	bondIDKey := bondKey(assetID, bondCoinID)
+	if !auth.registerBondWaiter(bondIDKey) {
+		// Waiter already running! They'll get a response to their first
+		// request, or find out on connect if the bond was activated.
+		return msgjson.NewError(msgjson.BondAlreadyConfirmingError, "bond already submitted")
+	}
+
+	var refundAddr string
+	if legacyFeePaid {
+		refundAddr = postBond.LegacyFeeRefundAddr
+	}
+
+	if confs >= reqConfs {
+		// No need to call checkFee again in a waiter.
+		postBondRes.Active = true
+		auth.storeBondAndRespond(conn, dbBond, acct, newAcct, msg.ID, postBondRes, refundAddr)
+		auth.removeBondWaiter(bondIDKey) // after storing it
+		return nil
+	}
+
+	auth.latencyQ.Wait(&wait.Waiter{
+		Expiration: time.Now().Add(time.Hour),
+		TryFunc: func() wait.TryDirective {
+			res := auth.waitBondConfs(conn, dbBond, acct, reqConfs, newAcct, msg.ID, postBondRes, refundAddr)
+			if res == wait.DontTryAgain {
+				auth.removeBondWaiter(bondIDKey)
+			}
+			return res
+		},
+		ExpireFunc: func() {
+			auth.removeBondWaiter(bondIDKey)
+			// User may retry postbond periodically or on reconnect.
+		},
+	})
+	// NOTE: server restart cannot restart these waiters, so user must
+	// resubmit their postbond after their request times out.
+
+	return nil
+}
+
+func (auth *AuthManager) storeBondAndRespond(conn comms.Link, bond *db.Bond, acct *account.Account,
+	newAcct bool, reqID uint64, postBondRes *msgjson.PostBondResult, refundAddr string) {
+	acctID := acct.ID
+	assetID, coinID := bond.AssetID, bond.CoinID
+	bondStr := coinIDString(assetID, coinID)
+	bondAssetSym := dex.BipIDSymbol(assetID)
+	var err error
+	if newAcct {
+		log.Infof("Creating new user account %v from %v, posted first bond in %v (%s)",
+			acctID, conn.Addr(), bondStr, bondAssetSym)
+		err = auth.storage.CreateAccountWithBond(acct, bond)
+	} else {
+		log.Infof("Adding bond for existing user account %v from %v, with bond in %v (%s)",
+			acctID, conn.Addr(), bondStr, bondAssetSym)
+		err = auth.storage.AddBond(acct.ID, bond)
+	}
+	if err != nil {
+		log.Errorf("Failure while storing bond for acct %v (new = %v): %v", acct, newAcct, err)
+		conn.SendError(reqID, &msgjson.Error{
+			Code:    msgjson.RPCInternalError,
+			Message: "failed to store bond",
+		})
+		return
+	}
+
+	if refundAddr != "" {
+		log.Warnf("Account %v posting bond requests a legacy fee refund to %v",
+			acctID, refundAddr) // signature checked above
+		// To actually do this, the operator would need to manually send funds
+		// to the address, then use an admin endpoint to somehow delete the old
+		// fee coin ID from the table.
```

Comment:
I don't think this legacy fee refund business is worth it any more, regardless of the level of automation.